### PR TITLE
Prediction tests

### LINKED ISSUE #4: [public api] Create tests for /v2/prediction
Create tests for /v2/prediction with pytest

### PR DESCRIPTION
**Pull request by RusticPotatoes**
For item https://github.com/orgs/Bot-detector/projects/14/views/1?pane=issue&itemId=40915254 [public api] Create tests for /v2/prediction

1. Adding test cases for prediction.py to satisfy properties testing. 

- Invalid Users
- Valid Users
- Invalid Users with Breakdown
- Valid Users with Breakdown 
- Extra test cases created with hypothesis.strategies text to generate 1) invalid player names from min 0-2 and 2) max 13-inf.

2. Updating Make File so Make Test shows more verbose output

3. hypothesis version frozen in requirements

### CODE REVIEW

File: tests/test_player_api.py [hunk 1]
```diff
@@ -8,23 +8,22 @@
...
-		error = f"Invalid response return type, expected list[dict]"
-		print(response.json())
-		assert isinstance(response.json(), list), error
\ No newline at end of file
+    def testInvalidPlayerBodyIsEmpty(self):
+        url = "http://localhost:5000/v2/players/score"
+        params = {"name": [""]}
+        response = requests.get(url, params)
+        error = f"Invalid response return type, expected list[dict]"
+        print(response.json())
```
Comment (extreme4all) [inline tests/test_player_api.py]: print statement from debugging?

**Review by extreme4all — CHANGES_REQUESTED** [review]
Please change the function names to snake_case and avoid numbers in the function names, everything else looks good!

maybe something else but in the documentation can there be something for testing on linux, "how to use the make file "?

**Comment by RusticPotatoes** [comment]
> Please change the function names to snake_case and avoid numbers in the function names, everything else looks good!
> 
> maybe something else but in the documentation can there be something for testing on linux, "how to use the make file "?

these should be both satisfied now. please review

File: tests/test_player_api.py (hunk 1 above)
Comment (RusticPotatoes) [reply tests/test_player_api.py]: Yea good catch. Removed it

File: README.md [hunk 2]
```diff
@@ -17,16 +17,112 @@ pip install -r requirements.txt
...
+### setup your enviornment
+prereqs: 
+- docker
+- python3
+- local clone of repo
+- terminal opened to cloned repos root path
+
+run `setup`
 ```sh
-python3 -m venv .venv
```
Comment (extreme4all) [inline README.md:25]: i jus noticed none of this is running in a venv? either we tell them to create a venv or we create a venv?
e.g python -m venv .venv

File: README.md [hunk 3]
```diff
@@ -17,16 +17,112 @@ pip install -r requirements.txt
...
 ## linux
-make sure to have pip
+use the `make` command with an `action` when developing on linux.  think of `actions` as a list of predefined commands to help simplify common development routines.
+
+#### view avaiable actions
+`make help` to see all avaiable actions
+
 ```sh
-sudo apt install python3-pip
+public-api$ make help
```
Comment (extreme4all) [inline README.md]: error occured
```
Command 'make' not found, but can be installed with:
sudo apt install make        # version 4.3-4.1build1, or
sudo apt install make-guile  # version 4.3-4.1build1
```

Comment (RusticPotatoes) [reply README.md]: ohh the make command may not be included in all gnu gcc pacakges.

File: README.md (hunk 2 above)
Comment (RusticPotatoes) [reply README.md:25]: yea we can put it in a venv.  i dont like managing them when i end up having multiple virtual environments for python.  little bit of overhead to figure out what venvs are available, what project is using which and cleaning up old ones that you dont use.  

if we do maybe we should name it project specific venv name.  like venv-public-api.

File: README.md (hunk 3 above)
Comment (RusticPotatoes) [reply README.md]: i added an entire section for the make package install instructions. should be resolved now

File: README.md (hunk 2 above)
Comment (RusticPotatoes) [reply README.md:25]: this should be resolved now, added to read me step to create venv and activate it

**Comment by extreme4all** [comment]
looks good!

**Pull request merged**